### PR TITLE
Do not raise on missing status blob; reduce amount of logging for vms…

### PR DESCRIPTION
**Pull request by narrieta**
The status blob url should not be a required property in vmSettings. Instead of raising an error, not it is initialized to None (to preserve the behavior established by extensionsConfig).

Also, since some errors are more common than expected, reduced the max number of errors from 5/hour to 3/hour (telemetry) and 1/hour (local log), and reduced the amount of details in parsing errors.

### CODE REVIEW

File: tests/data/wire/ext_conf_no_extensions.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<Extensions version="1.0.0.0" goalStateIncarnation="9">
```
Comment (nagworld9) [inline tests/data/wire/ext_conf_no_extensions.xml:1]: where is this file used?

File: azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py [hunk 2]
```diff
@@ -173,13 +173,15 @@ def _parse_status_upload_blob(self, vm_settings):
...
-        if self._status_upload_blob is None:
-            raise Exception("Missing statusUploadBlob.statusBlobType")
+            self._status_upload_blob = None
+            self._status_upload_blob_type = "BlockBlob"
+        else:
+            self._status_upload_blob = status_upload_blob.get("value")
+            if self._status_upload_blob is None:
+                raise Exception("Missing statusUploadBlob.value")
+            self._status_upload_blob_type = status_upload_blob.get("statusBlobType")
+            if self._status_upload_blob is None:
```
[the remark this answers is not in the document]
Comment (nagworld9) [reply azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py:183]: can you add failure scenario if you could?

[the remark this answers is not in the document]
Comment (narrieta) [reply azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py:183]: thanks; fixed typo. the test already exists: test_create_from_vm_settings_should_assume_block_when_blob_type_is_not_valid

File: tests/data/wire/ext_conf_no_extensions.xml (hunk 1 above)
Comment (narrieta) [reply tests/data/wire/ext_conf_no_extensions.xml:1]: removed

File: azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nagworld9) [reply azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py]: Actually, this typo carried from develop and mentioned test case not covering None case that's why we didn't catch it before.

[the remark this answers is not in the document]
Comment (nagworld9) [reply azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py]: Approving it unless you want to add blob_type is None case, thanks

**Review by nagworld9 — APPROVED** [review]

File: azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (narrieta) [reply azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py]: checking for invalid is good enough. it is unrealistic to check for every possible invalid value: missing, empty string, mixed case, garbage, etc. 

the existing test captures the desired behavior: an invalid value defaults to Block

[the remark this answers is not in the document]
Comment (nagworld9) [reply azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py]: Not agreed 100%, At least we should have a test case that was handled it in the code. My intention is we would have caught this typo if we have test case where status_upload_blob_type is None (not some invalid value) and assert BlockBlob type.

[the remark this answers is not in the document]
Comment (narrieta) [reply azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py]: Yes, we would have captured this typo, but we did not know we were going to make this typo. As another example: should we add a test for mixed case that could catch a possibly defective piece of code that doesn't handle that case?

[the remark this answers is not in the document]
Comment (nagworld9) [reply azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py]: I agree but my only point in general, we were expected certain behavior when it is None. That behavior should have been tested. 

Beyond this, we fixed the typo I'm good with that :)

**Pull request merged**